ukol2/ukol2.py

```diff
@@ -1,67 +1,71 @@
 import requests
 
 
-def vyhledat_podnikatelsky_subjekt_ico():
-    ico = input("Zadejte ICO: ")
+def search_subject(ico: str) -> str:
     url = f"https://ares.gov.cz/ekonomicke-subjekty-v-be/rest/ekonomicke-subjekty/{ico}"
     response = requests.get(url)
 
     if response.status_code == 200:
         data = response.json()
         obchodni_jmeno = data['obchodniJmeno']
         adresa = data['sidlo']['textovaAdresa']
-        print(f"{obchodni_jmeno}\n{adresa}")
-    else:
-        print("Nepodarilo se najit subjekt s danym ICO.")
+        return f"{obchodni_jmeno}\n{adresa}"
+    return f"Nepodarilo se najit subjekt s danym ICO: {ico}. Kód chyby: {response.status_code}"
 
-vyhledat_podnikatelsky_subjekt_ico()
 
-
-def stahnout_ciselnik_pravnich_forem():
+def get_code_list() -> list:
     url = "https://ares.gov.cz/ekonomicke-subjekty-v-be/rest/ciselniky-nazevniky/vyhledat"
+
     headers = {
         "accept": "application/json",
         "Content-Type": "application/json",
     }
-    data = '{"kodCiselniku": "PravniForma", "zdrojCiselniku": "res"}'
-    response = requests.post(url, headers=headers, data=data)
+
+    data = {"kodCiselniku": "PravniForma", "zdrojCiselniku": "res"}
+    response = requests.post(url, headers=headers, json=data)
 
     if response.status_code == 200:
         data = response.json()
         return data['ciselniky'][0]['polozkyCiselniku']
-    else:
-        print("Nepodařilo se stáhnout číselník právních forem.")
-        return {}
```

Comment:
toto je hodně neprogramátorské :-) Obecně se v kurzu neučíme moc to, jak by měly vypadat vrácené hodnoty funkcí, avšak obecně print() ve funkcích není moc vhodný a vrácení prázdného dict také ne. 

Upravil jsem celkově funkci tak, že na začátku je definované, že vrací list, tedy ať již při úspěchu (return data['ciselniky'][0]['polozkyCiselniku'] - toto je rovnou list) nebo neúspěchu return [f"Nepodařilo se stáhnout číselník právních forem. Kód odpovědi: {response.status_code}"] vrátí vždy list